trace/opencensus/trace/tracer/context_tracer.py

```diff
@@ -59,6 +61,7 @@ def span(self, name='span'):
         :returns: The Span object.
         """
         span = self.start_span(name=name)
+        span.__class__.__exit__ = self.end_span
```

Comment:
I don't quite understand this.  Why are we modifying `span.__class__` at runtime?  Doesn't the existing `Span.__exit__()` call `ContextTracer.end_span()`?

Also, in `Span.__exit__`, I wonder if we should be calling `self.finish()` before calling `end_span()`, instead of after?

Reply:
I was removing the `ContextTracer` from span class and modifying the span.__class__.__exit__ to use the `ContextTracer.end_span()` instead. Changed back to the previous implementation as it's not that safe to modify the span.__class__.
And for `span.__exit__`, if it's calling `ContextTracer.end_span()`, the `span.finish()` is called inside `tracer.end_span()` method and then return, the finish function won't be executed twice.